Sources/Sentry/SentryNSProcessInfoWrapper.mm

```diff
@@ -0,0 +1,50 @@
+#import "SentryNSProcessInfoWrapper.h"
+#import "SentryDependencyContainer.h"
+#import "SentryNSNotificationCenterWrapper.h"
+
+@implementation SentryNSProcessInfoWrapper
+
+- (NSProcessInfoThermalState)thermalState
+{
+    return NSProcessInfo.processInfo.thermalState;
+}
+
+- (BOOL)isLowPowerModeEnabled
+{
+    return NSProcessInfo.processInfo.isLowPowerModeEnabled;
+}
+
+- (void)monitorForPowerStateChanges:(id)target callback:(SEL)callback
+{
+    // According to Apple docs: "This notification is posted on the global dispatch queue. The
```

Comment:
`l`: It would be great to provide the link to the apple docs.

Reply:
I can point to NSProcessInfo.h, see my reply to your next comment: this isn't actually in the online docs :/